src/test/java/de/unistuttgart/iste/gits/content_service/service/ContentServiceTest.java

```diff
@@ -0,0 +1,143 @@
+package de.unistuttgart.iste.gits.content_service.service;
+
+import de.unistuttgart.iste.gits.content_service.persistence.dao.ContentEntity;
+import de.unistuttgart.iste.gits.content_service.persistence.dao.TagEntity;
+import de.unistuttgart.iste.gits.content_service.persistence.repository.ContentRepository;
+import de.unistuttgart.iste.gits.content_service.persistence.repository.TagRepository;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+public class ContentServiceTest {
+    @Mock
+    private ContentRepository contentRepository;
+    @Mock
+    private TagRepository tagRepository;
+    @InjectMocks
+    private ContentService contentService;
+
+    @Test
+    void testDeleteContentSuccessful() {
+        UUID testContentId = UUID.randomUUID();
+        doNothing().when(contentRepository).deleteById(any());
+        doReturn(true).when(contentRepository).existsById(any());
+        UUID deletedContentId = contentService.deleteContent(testContentId);
+        assertThat(deletedContentId,is(testContentId));
+        verify(contentRepository).deleteById(testContentId);
+    }
+
+    Set<String> getTagNames(Collection<TagEntity> tags) {
+        return tags.stream().filter(tag -> tag != null).map(tag -> tag.getName()).collect(Collectors.toSet());
+    }
+
+    @Test
+    void testTagSynchronisationNoExistingTagsSuccessful() {
```

Comment:
If synchronize is done in separate class it can also be tested in seperate class